R/LSTconfidenceIntervals.R

```diff
@@ -0,0 +1,399 @@
+  
+  .plotDatasets(confidenceContainer, options)
+  
+  .plotConvergencePlot(confidenceContainer, options)
+  
+  return()
+}
+
+.getConfidenceContainer <- function(jaspResults) {
+  confidenceContainer <- jaspResults[["confidenceContainer"]] %setOrRetrieve% createJaspContainer(dependencies = c("mu", "sigma", "n", "confidenceIntervalInterval", "nReps", "treePlot"))
+  return(confidenceContainer)
+}
+
+.simulateDatasets <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["simulatedDatasets"]]))
+    return()
+  
+  listWithData <- lapply(seq_len(options[["nReps"]]), function(x) rnorm(options$n,
+                                                            options$mu,
+                                                            options$sigma))
+  
+  jaspContainer[["simulatedDatasets"]] <- createJaspState(object = listWithData)
+  
+  return()
+}
+
+.computeConfidenceIntervals <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["computedConfidenceIntervals"]]))
+    return()
+  
+  listWithData <- jaspContainer[["simulatedDatasets"]][["object"]]
+  
+  meanCI <- t(sapply(listWithData, function(x) c(mean(x), 
+                                                 t.test(x, conf.level = options$confidenceIntervalInterval)$conf.int[1:2])))
+  meanCI <- as.data.frame(meanCI)
+  colnames(meanCI) <- c("mean", "lower", "upper")
+  meanCI[["Repetition"]] <- 1:options$nReps
+  meanCI[["successfulCI"]] <- options$mu > meanCI[["lower"]] & options$mu < meanCI[["upper"]]
+  meanCI[["colorCI"]] <- ifelse(meanCI[["successfulCI"]], "A", "B")
+  meanCI[["sd"]] <- sapply(listWithData, sd)
+  meanCI[["n"]] <- sapply(listWithData, length)
+  
+  
+  jaspContainer[["computedConfidenceIntervals"]] <- createJaspState(object = meanCI)
+  
+  return()
+}
+
+.makeSummaryTable <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["resultTable"]]))
+    return()
+  
+  resultTable <- createJaspTable(title = "Results Table")
+  jaspContainer[["resultTable"]] <- resultTable
+  
+  resultTable$addColumnInfo(name = "Repetitions", type = "integer")
+  resultTable$addColumnInfo(name = "mu", type = "number", title=gettext("μ"))
+  resultTable$addColumnInfo(name = "sigma", title=gettext("σ"), type = "number")
+  resultTable$addColumnInfo(name = "n", type = "integer")
+  resultTable$addColumnInfo(name = "coverage", type = "integer",title=gettext("Coverage"))
+  resultTable$addColumnInfo(name = "coveragePercentage", type = "number", format = "pc", title=gettext("Coverage %"))
+  
+  meanCI <- jaspContainer[["computedConfidenceIntervals"]][["object"]]
+  tableDat <- data.frame(mu = options$mu,
+                         sigma = options$sigma,
+                         Repetitions = options$nReps,
+                         n = options$n,
+                         coverage = sum(meanCI[["successfulCI"]]),
+                         coveragePercentage = mean(meanCI[["successfulCI"]])
+  
+  resultTable$setData(tableDat)
```

Comment:
not sure why the unit tests don't complain about this, but shouldn't there be a `)` here that closes the `data.frame(`? The last one closes `mean(meanCI...`

Reply:
hahaha yeah very weird (and sloppy..); I just closed the bracket.